File: backend/channels/whatsapp/whatsapp_api.js
const express = require('express');
const bodyParser = require('body-parser');
const WhatsAppCloudAPI = require('whatsappcloudapi_wrapper');
require('dotenv').config();

const app = express();
app.use(bodyParser.json());

const whatsapp = new WhatsAppCloudAPI({
    accessToken: process.env.WHATSAPP_API_TOKEN,
    senderPhoneNumberId: process.env.WHATSAPP_PHONE_NUMBER_ID,
});

app.post('/send_message', async (req, res) => {
    const { to, content } = req.body;
    try {
        const response = await whatsapp.sendText({
            recipientPhone: to,
            message: content,
        });
        res.status(200).json(response);
    } catch (error) {
        res.status(500).json({ error: error.message });
    }
});

app.post('/send_otp', async (req, res) => {
    const { to, otp } = req.body;
    const content = `Your OTP is ${otp}`;
    try {
        const response = await whatsapp.sendText({
            recipientPhone: to,
            message: content,
        });
        res.status(200).json(response);
    } catch (error) {
        res.status(500).json({ error: error.message });
    }
});

const PORT = process.env.PORT || 8082;
app.listen(PORT, () => {
    console.log(`Server is running on port ${PORT}`);
});

File: backend/channels/whatsapp/whatsapp_api.py
import requests
import os
from fastapi import FastAPI
from pydantic import BaseModel
import uvicorn

class WhatsAppMessage(BaseModel):
    to: str
    content: str

class OTPMessage(BaseModel):
    to: str
    otp: str

app = FastAPI()

@app.post("/send_message/")
async def send_message(message: WhatsAppMessage):
    return send_whatsapp_message(message.to, message.content)

@app.post("/send_otp/")
async def send_otp(message: OTPMessage):
    content = f"Your OTP is {message.otp}"
    return send_whatsapp_message(message.to, content)

def send_whatsapp_message(to: str, content: str):
    url = f"https://graph.facebook.com/v17.0/{os.getenv('WHATSAPP_PHONE_NUMBER_ID')}/messages"
    headers = {
        "Authorization": f"Bearer {os.getenv('WHATSAPP_API_TOKEN')}",
        "Content-Type": "application/json"
    }
    payload = {
        "messaging_product": "whatsapp",
        "to": to,
        "type": "text",
        "text": {"body": content}
    }
    response = requests.post(url, json=payload, headers=headers)
    return response.json()

if __name__ == "__main__":
    uvicorn.run(app, host="0.0.0.0", port=8082)
